microprot/scripts/pdb_search.py

```diff
@@ -0,0 +1,393 @@
+        indication that the provided line is no HHsearch hit summary line.
+
+    [1] Johannes Soeding, "Quick guide to HHsearch"
+        ftp://ftp.tuebingen.mpg.de/pub/protevo/HHsearch/HHsearch1.5.01/HHsearc
+        h-guide.pdf
+    """
+
+    fields = line.rstrip().split()
+    hit = {}
+
+    try:
+        # Index of hit
+        hit[_HEADER[0]] = int(fields[0])
+
+        # First 30 characters of domain description (from nameline of query
+        # sequence)
+        hit[_HEADER[1]] = " ".join(fields[1:-9])
+
+        # Probability of target to be a true positive For the probability of
+        # being a true positive, the secondary structure score in column 7 is
+        # taken into account, together with the raw score in column 6
+        # (’Score’). True positives are defined to be either globally
+        # homologous or they are at least locally similar in structure. More
+        # precisely, the latter criterion demands that the MAXSUB score between
+        # query and hit is at least 0.1. In almost all cases the structural
+        # similarity will we be due to a global OR LOCAL homology between query
+        # and target.
+        hit[_HEADER[-9]] = float(fields[-9])
+
+        # Expect-value E-value and P-value are calculated without taking the
+        # secondary structure into account! The E-value gives the average
+        # number of false positives (’wrong hits’) with a score better than the
+        # one for the target when scanning the datbase. It is a measure of
+        # reliability: E-values near to 0 signify a very reliable hit, an
+        # E-value of 10 means about 10 wrong hits are expected to be found in
+        # the database with a score at least this good.
+        hit[_HEADER[-8]] = float(fields[-8])
+
+        # P-value The P-value is just the E-value divided by the number of
+        # sequences in the database. It is the probability that in a PAIRWISE
+        # comparison a wrong hit will score at least this good.
+        hit[_HEADER[-7]] = float(fields[-7])
+
+        # Raw score, does not include the secondary structure score
+        hit[_HEADER[-6]] = float(fields[-6])
+
+        # Secondary structure score This score tells how well the PSIPRED-
+        # predicted (3-state) or actual DSSP-determined (8-state) secondary
+        # structure sequences agree with each other. PSIPRED confidence values
+        # are used in the scoring, low confidences getting less statistical
+        # weight.
+        hit[_HEADER[-5]] = float(fields[-5])
+
+        # The number of aligned Match columns in the HMM-HMM alignment.
+        hit[_HEADER[-4]] = int(fields[-4])
+
+        # Range of aligned match states from query HMM
+        hit[_HEADER[-3]] = fields[-3]
+
+        # Range of aligned match states from target HMM
+        hit[_HEADER[-2]] = fields[-2]
+
+        # Number of match states in target HMM
+        hit[_HEADER[-1]] = int(fields[-1][1:-1])
+
+        return hit
+    except ValueError:
```

Comment:
ValueError is raised twice here. Will there be a better way to customize error message?